embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/features/ResurrectionFeatureTest.kt

```diff
@@ -79,6 +88,47 @@ internal class ResurrectionFeatureTest {
         )
     }
 
+    @Test
+    fun `empty crash envelope not available for native crash resurrection if background activity is enabled`() {
+        testRule.runTest(
+            persistedRemoteConfig = RemoteConfig(
+                backgroundActivityConfig = BackgroundActivityRemoteConfig(100f)
+            ),
+            setupAction = {
+                cacheStorageServiceProvider = { cacheStorageService }
+            },
+            testCaseAction = {
+                recordSession()
+                recordSession()
+            },
+            assertAction = {
+                getSessionEnvelopes(2)
+                assertTrue(cacheStorageService.getCachedCrashEnvelope().isEmpty())
+            }
+        )
+    }
+
+    @Test
+    fun `one empty crash envelope available for native crash resurrection if background activity is not enabled`() {
+        testRule.runTest(
+            persistedRemoteConfig = RemoteConfig(
+                backgroundActivityConfig = BackgroundActivityRemoteConfig(0f)
+            ),
+            setupAction = {
+                cacheStorageServiceProvider = { cacheStorageService }
+            },
+            testCaseAction = {
+                recordSession()
+                recordSession()
+                recordSession()
+            },
+            assertAction = {
+                getSessionEnvelopes(3)
+                assertEquals(1, cacheStorageService.getCachedCrashEnvelope().size)
```

Comment:
It's probably worth enhancing the assertions to check the contents of the envelope at a basic level

Reply:
Sure

Reply:
Actually was named confusingly - we don't actually care about the envelope that is persisted given that we fake the `PayloadStorageService`, so what we want to validate is only the metadata is correct, which I'll do.